addons/website/models/website_snippet_filter.py

```diff
@@ -268,30 +298,38 @@ def _get_hardcoded_sample(self, model):
         """
         return [{}]
 
-    def _filter_records_to_values(self, records):
+    def _filter_records_to_values(self, records, is_sample=False):
         """
         Extract the fields from the data source 'records' and put them into a dictionary of values
 
         @param records: Model records returned by the filter
+        @param is_sample: True if conversion if for sample records
 
         @return List of dict associating the field value to each field name
         """
-
         self.ensure_one()
         meta_data = self._get_filter_meta_data()
 
         values = []
-        model = self.env[self.filter_id.model_id]
+        model = self._get_model()
         Website = self.env['website']
         for record in records:
             data = {}
             for field_name, field_widget in meta_data.items():
                 field = model._fields.get(field_name)
-                if field.type == 'binary':
-                    data[field_name] = Website.image_url(record, field_name)
+                if field and field.type in ('binary', 'image'):
+                    if is_sample:
+                        if 'image_1920' in record and record.image_1920:
+                            data[field_name] = record.image_1920.decode('utf8')
+                        elif field_name in record and record[field_name]:
```

Comment:
why this special case ?

Reply:
This is indeed not needed, was from an intermediate state